server/resp/src/main/java/org/infinispan/server/resp/serialization/lua/LuaResponseWriter.java

```diff
@@ -0,0 +1,285 @@
+
+   public LuaResponseWriter(Lua lua) {
+      this.lua = lua;
+   }
+
+   @Override
+   public RespVersion version() {
+      return version;
+   }
+
+   @Override
+   public void version(RespVersion version) {
+      this.version = version;
+   }
+
+   @Override
+   public boolean isInternal() {
+      return true;
+   }
+
+   @Override
+   public void nulls() {
+      lua.checkStack(1);
+      lua.pushNil();
+   }
+
+   @Override
+   public void ok() {
+      lua.checkStack(1);
+      lua.push("OK");
+   }
+
+   @Override
+   public void queued(Object ignore) {
+      throw new UnsupportedOperationException("queued(Object)");
+   }
+
+   @Override
+   public void simpleString(CharSequence value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value.toString());
+      }
+   }
+
+   @Override
+   public void string(CharSequence value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value.toString());
+      }
+   }
+
+   @Override
+   public void string(byte[] value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(new String(value, StandardCharsets.US_ASCII));
+      }
+   }
+
+   @Override
+   public void integers(Number value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value);
+      }
+   }
+
+   @Override
+   public void doubles(Number value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+      lua.push(value);
+      }
+   }
+
+   @Override
+   public void booleans(boolean value) {
+      lua.checkStack(1);
+      lua.push(value);
+   }
+
+   @Override
+   public void arrayEmpty() {
+      lua.checkStack(1);
+      lua.push(false);
+   }
+
+   @Override
+   public <T> void array(Collection<T> collection, JavaObjectSerializer<T> serializer) {
+      if (collection == null) {
+         nulls();
+      } else {
+         lua.checkStack(2);
+         lua.newTable();
+         int i = 1;
+         for (T o : collection) {
+            lua.push(i);
+            serializer.accept(o, this);
+            lua.setTable(-3);
+            i++;
+         }
+      }
+   }
+
+   @Override
+   public void array(Collection<?> collection, Resp3Type contentType) {
+      if (collection == null) {
+         nulls();
+      } else {
+         lua.checkStack(2);
+         lua.newTable();
+         int i = 1;
+         for (Object o : collection) {
+            lua.push(i);
+            contentType.serialize(o, this);
+            lua.setTable(-3);
+            i++;
+         }
+      }
+   }
+
+   @Override
+   public void emptySet() {
+      throw new UnsupportedOperationException("emptySet"); // Only used by COMMAND which cannot be called from Lua.
```

Comment:
Is it possible we add other commands that may support this though? I fear no one will update this and the command added won't be tested by a lua script we are testing. Maybe we should still implement or is it difficult to do so?

Reply:
Probably, but that means creating tests just to exercise parts of the code that won't ever be executed at the moment